WCFDataService/Client/System/Data/Services/Client/Binding/DataServiceCollectionOfT.cs

```diff
@@ -758,15 +758,29 @@ private void InternalLoadCollection(IEnumerable<T> items)
             Debug.Assert(!(items is DataServiceQuery), "SL Client using DSQ as items...should have been caught by ValidateIteratorParameter.");
 #endif
 
+            if (this.IsTracking && this.observer != null)
```

Comment:
Further more, can you share the code you used to verify those changes? I can migrate them to our internal test.